domain/application/state/storage.go

```diff
@@ -140,6 +140,30 @@ type storageTemplate struct {
 	params application.ApplicationStorageArg
 }
 
+func (st *State) composeStorageTemplates(ctx context.Context, tx *sqlair.TX, appUUID coreapplication.ID, args []application.ApplicationStorageArg) ([]storageTemplate, error) {
+	templates := make([]storageTemplate, 0, len(args))
+	for _, arg := range args {
+		storageMeta, err := st.getApplicationCharmStorageByName(ctx, tx, appUUID, arg.Name)
+		if errors.Is(err, charmStorageNotFound) {
+			return nil, errors.Errorf(
+				"charm for application %q has no storage called %q",
+				appUUID, arg.Name,
+			).Add(applicationerrors.StorageNameNotSupported)
+		} else if err != nil {
+			return nil, errors.Errorf("getting charm storage metadata for storage name %q application %q: %w", arg.Name, appUUID, err)
+		}
+
+		if arg.Count == 0 {
+			continue
+		}
```

Comment:
Why is this done after getting the storageMeta? Seems like we could do this first.

Reply:
Interesting.